### DISCUSSION TITLE
Character caching

### DISCUSSION THREAD

**Discussion by potsmugen** [ikemen-engine/Ikemen-GO#1276] [Feature Request] [answered]
### Is your feature request related to a problem? Please describe.

In Mugen, if you pick the same character for more than one player, the engine will only load the character once. The clipboard even tells you this in 1.1:

![Image1](https://github.com/ikemen-engine/Ikemen-GO/assets/107247004/524f3346-0098-45dc-9df0-18b812ffb9b9)

In Ikemen GO, the characters will be loaded redundantly, increasing load times for larger characters and (I suppose) memory usage.

### Describe the solution you'd like

Character caching as in Mugen. This seems to already happen when you exit a match and play a new match with the same characters in the same spots, so it might be a matter of expanding that functionality.

**Comment by Lazin3ss** [comment]
This was attempted in the past, but there were problems with palettes so it was reverted. We should further improve resource management in the future.

**Reply by potsmugen** [reply]
I think it's a bit different from how SFF caching was implemented. Like, characters are currently cached according to their def files or so, so even if they use the same SFF it will be loaded separately and the palette overlaps don't happen. I was wondering if the current behavior of caching the chars for each player (for rematches) could be extended to caching them for any player.

**Comment by potsmugen** [comment, marked as answer]
This already has a stable implementation. Closing.
